src/main/java/Luke.java

```diff
@@ -0,0 +1,246 @@
+        for (String validCommand: validCommands) {
+            if (command.equals(validCommand)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public static void main(String[] args) {
+
+        String name = "Luke";
+
+        ArrayList<Task> taskList = new ArrayList<Task>();
+
+        ArrayList<String> saveTaskList = new ArrayList<String>();
+
+        int noTasks = 0;
+
+        try {
+            File directory = file.getParentFile();
+            if (!directory.exists()) {
+                directory.mkdirs();
+            }
+
+            if (!file.exists()) {
+                file.createNewFile();
+            }
+        } catch (IOException e) {
+            System.out.println("Failed to save file: " + e.getMessage());
+        }
+
+        try {
+            Scanner myReader = new Scanner(file);
+            while (myReader.hasNextLine()) {
+                String data = myReader.nextLine();
+                if (data != "") {
+                    saveTaskList.add(data);
+                    noTasks++;
+                }
+            }
+            myReader.close();
+            fileIsEmpty = saveTaskList.isEmpty();
+        } catch (FileNotFoundException e) {
+            System.out.println("File not found.");
+        }
+
+        saveTaskList.forEach(taskString -> {
+           String taskType = taskString.substring(1, 2);
+           switch (taskType) {
+               case "T":
+                   Todo todo = new Todo(taskString.substring(7).trim());
+                   taskList.add(todo);
+                   break;
+               case "D":
+                   String[] deadlineSplit = taskString.split("by: ");
+                   String deadlineDescription = deadlineSplit[0].substring(7, deadlineSplit[0].length() - 1);
+                   String by = deadlineSplit[1].substring(0, deadlineSplit[1].length() - 1).trim();
+                   Deadline deadline = new Deadline(deadlineDescription, by);
+                   taskList.add(deadline);
+                   break;
+               case "E":
+                   String[] eventFirstSplit = taskString.split("from: ");
+                   String[] eventSecondSplit = eventFirstSplit[1].split(" to: ");
+                   String eventDescription = eventFirstSplit[0].substring(7, eventFirstSplit[0].length() - 1);
+                   String from = eventSecondSplit[0].trim();
+                   String to = eventSecondSplit[1].substring(0, eventSecondSplit[1].length() - 1).trim();
+                   Event event = new Event(eventDescription, from, to);
+                   taskList.add(event);
+                   break;
+           }
+        });
+
+        System.out.println("Hello! I'm " + name + "\nWhat can I do for you?");
+
+        Scanner scanner = new Scanner (System.in);
+        String input = scanner.nextLine();
+        while (!input.equals("bye")) {
+            String[] splited = input.split(" ");
```

Comment:
Perhaps can describe it a bit more rather than using splited as a variable name? (by the way, it should be split).